### PR TITLE
Start visit

### PR DESCRIPTION
**Pull request by nkimaina**

### CODE REVIEW

File: src/utils/use-session-user.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default function useSessionUser() {
+  const [sessionUser, setSessionUser] = useState<any>(null);
+  useEffect(() => {
+    let currentUserSub: Subscription;
+    if (sessionUser === null) {
+      openmrsObservableFetch("/ws/rest/v1/session").subscribe(user => {
+        setSessionUser(user.data);
+      });
+    }
+    return () => currentUserSub.unsubscribe();
```
Comment (FlorianRappl) [inline src/utils/use-session-user.tsx]: This will fail - currentUserSub may (i.e., will) be undefined

File: src/utils/use-session-user.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { useState, useEffect } from "react";
+import { Subscription } from "rxjs";
+import { openmrsObservableFetch } from "@openmrs/esm-api";
+
+export default function useSessionUser() {
+  const [sessionUser, setSessionUser] = useState<any>(null);
+  useEffect(() => {
+    let currentUserSub: Subscription;
+    if (sessionUser === null) {
+      openmrsObservableFetch("/ws/rest/v1/session").subscribe(user => {
```
Comment (FlorianRappl) [inline src/utils/use-session-user.tsx]: I think you want to set `currentUserSub` here, right?

File: src/widgets/location/location-select.component.tsx [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+export default function LocationSelect(props: LocationSelectProps) {
+  const [locations] = useLocations();
+  const onLocationsChanged = event => {
+    props.onLocationChanged(
+      locations.find(loc => loc.uuid == event.target.value)
+    );
+  };
+  return (
+    <>
```
Comment (FlorianRappl) [inline src/widgets/location/location-select.component.tsx]: Is the fragment really necessary in this case? There is just one child element.

File: src/widgets/location/location-select.component.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      <select
+        name="visitLocation"
+        id={props.id || "visitLocation"}
+        className="omrs-type-body-regular"
+        style={{ height: "40px" }}
+        value={props.currentLocationUuid}
+        onChange={onLocationsChanged}
+      >
+        <option value={null} className="omrs-padding-8">
+          {""}
```
Comment (FlorianRappl) [inline src/widgets/location/location-select.component.tsx]: I think this is the same as `<option value="" className="omrs-padding-8" />`

File: src/widgets/location/use-locations.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+import { Subscription } from "rxjs";
+import { Location, getLocations } from "./location.resource";
+
+export default function useLocations() {
+  const [locations, setLocations] = useState<Array<Location>>([]);
+  useEffect(() => {
+    let locationSub: Subscription;
```
Comment (FlorianRappl) [inline src/widgets/location/use-locations.tsx]: Why not just `const locationSub = ...`? I think the typing would also benefit (no need for the explicit `Subscription`)

File: src/widgets/visit/use-started-visit.tsx [hunk 6]
```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from "react";
+import { Subscription } from "rxjs";
+import dayjs, { Dayjs } from "dayjs";
+import { getCurrentPatientUuid } from "@openmrs/esm-api";
+
+import { getVisitsForPatient } from "./visit.resource";
+
+export default function useStartedVisit(props: StartedVisitProps) {
+  const [startedVisit, setStartedVisit] = useState(undefined);
```
Comment (FlorianRappl) [inline src/widgets/visit/use-started-visit.tsx]: `useState<?>(undefined)` - what is it?

File: src/widgets/visit/use-visit-types.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { useState, useEffect } from "react";
+import { Subscription } from "rxjs";
+import { VisitType, getVisitTypes } from "./visit-type.resource";
+
+export default function useVisitTypes() {
+  const [visitTypes, setVisitTypes] = useState<Array<VisitType>>([]);
+  useEffect(() => {
+    let visitTypesSub: Subscription;
+    visitTypesSub = getVisitTypes().subscribe(
```
Comment (FlorianRappl) [inline src/widgets/visit/use-visit-types.tsx]: See above - just assign in initialization.

File: src/widgets/visit/visit-type-select.component.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    <>
+      <select
+        name="visitType"
+        id={props.id || "visitType"}
+        className="omrs-type-body-regular"
+        style={{ height: "40px" }}
+        onChange={onVisitTypesChanged}
+      >
+        <option value={null} className="omrs-padding-8">
+          {""}
```
Comment (FlorianRappl) [inline src/widgets/visit/visit-type-select.component.tsx]: Same as above.

**Review by FlorianRappl — CHANGES_REQUESTED** [review]
I think we are on a great track here. Thanks for the work!

**Comment by nkimaina** [comment]
@FlorianRappl Sorry that these changes are a lot to review, we are transferring this functionality from an internal repo. Also, we've not had a design session to agree on how visits should function, but these are standard features that will most likely be needed when we have the design.

**Review by donaldkibet — APPROVED** [review]
lgtm

**Pull request merged**